src/js/Field.js

```diff
@@ -128,6 +128,14 @@
                 this.showingDefaultData = true;
             }
 
+            if(!Alpaca.isValEmpty(this.data) &&
+                !Alpaca.isEmpty(this.options["prefix"]) &&
+                !Alpaca.startsWith(this.data, this.options["prefix"])){
+                this.data = `${this.options["prefix"]}${this.data}`
```

Comment:
We have an `.eslintrc` at the root of the DocSpring repo, and this should run some linting rules on any JavaScript code in the `app` or `vendor` directory. It's configured with `"plugins": ["es5"]`, which excludes any new ES6 syntax. Unfortunately this is needed because we are still using Sprockets with the Rails asset pipeline, and Sprockets doesn't support any newer JavaScript syntax, such as the backticks for interpolation `` `...` ``. This is normally not an issue in modern browsers, but it can cause issues in browsers that don't support newer features, since we don't have any [BabelJS](https://babeljs.io/) processor setup for browser compatibility.

So we just have to be careful about only using the old ES5 JavaScript for any JS in our Rails app assets. That means that this line should be changed to standard ES5 JavaScript:

```
this.data = this.options["prefix"] + this.data
```

Reply:
But also I'm not too sure if this is right place for this code. I think `prefix` is only used in the `CurrencyField`, and is not used by `Field` or any other descendants. So I would probably put this in the `constructor` of `CurrencyField`, underneath the call to `this.base(...`

Reply:
Another thing to note is that `prefix` is an option that is passed to the [jQuery Price Format plugin](https://github.com/price-format/Jquery-Price-Format), during `$(field).priceFormat(self.options);`. I'm still not sure what's going wrong with this library, but I would like to investigate further and perhaps find a cleaner solution